python_modules/dagster/dagster/_config/structured_config.py

```diff
@@ -92,9 +100,11 @@ def __setattr__(self, name: str, value: Any):
 
         return super().__setattr__(name, value)
 
-    def resource_function(self, context) -> Any:  # pylint: disable=unused-argument
-        # Default behavior, for "new-style" resources, is to return the resource itself, so that
-        # initialization is a no-op
+    def create_object_to_pass_to_user_code(self, context) -> Any:  # pylint: disable=unused-argument
+        # This acts identically to the function decorator with @resource in the old style
+        #
+        # Default behavior, for "new-style" resources, is to return itself, passing
+        # the actual resource object to user code.
```

Comment:
may as well make this an actual docstring

```suggestion
    def create_object_to_pass_to_user_code(self, context) -> Any:  # pylint: disable=unused-argument
        """
        Returns the object that this resource hands to user code, accessible by ops or assets
        through the context or resource parameters. This works like the function decorated
        with @resource when using function-based resources.
        
        Default behavior for new class-based resources is to return itself, passing
        the actual resource object to user code.
        """
```